src/BackupChain.cs

```diff
@@ -66,54 +75,35 @@ private static IEnumerable<BackupMetadata> MostRecentFullBackup(IEnumerable<Back
       if(!fullBackupsOrdered.Any())
         throw new BackupChainException("Could not find any full backups");
 
-      var targetCheckpointLsn = fullBackupsOrdered.First().CheckpointLsn;
       // get all the stripes of this backup
-      return fullBackupsOrdered.Where(fullBackup => fullBackup.CheckpointLsn == targetCheckpointLsn);
+      return fullBackupsOrdered.First();
     }
 
-    private static IEnumerable<BackupMetadata> MostRecentDiffBackup(IEnumerable<BackupMetadata> backups,
+    private static StripedBackupSet MostRecentDiffBackup(IEnumerable<StripedBackupSet> backups,
       BackupMetadata lastFullBackup)
     {
-      var diffBackupsOrdered = backups
-        .Where(b =>
-                 b.BackupType == BackupFileTools.BackupType.Diff &&
-                 b.DatabaseBackupLsn == lastFullBackup.CheckpointLsn)
-        .OrderByDescending(b => b.LastLsn).ToList();
-
-      if(!diffBackupsOrdered.Any())
-        return new List<BackupMetadata>();
-
-      var targetLastLsn = diffBackupsOrdered.First().LastLsn;
-      // get all the stripes of this backup
-      return diffBackupsOrdered.Where(diffBackup => diffBackup.LastLsn == targetLastLsn);
+      return backups.Where(b => b.BackupType == BackupFileTools.BackupType.Diff &&
+                                b.DatabaseBackupLsn == lastFullBackup.CheckpointLsn)
+                    .OrderByDescending(b => b.LastLsn)
+                    .FirstOrDefault();

```

Comment:
I like what you did with the others - removing the redundant `Where()` and putting the predicate logic in the `FirstOrDefault()`.
Let's keep that consistent, do you mind changing that here as well pls?